wgpu-core/src/command/memory_init.rs

```diff
@@ -285,73 +261,36 @@ impl<A: hal::Api> BakedCommands<A> {
-                        .ok_or(DestroyedTextureError(texture_use.id))?;
-
-                    let mut texture_barriers = Vec::new();
-                    let mut zero_buffer_copy_regions = Vec::new();
-                    for range in &ranges {
-                        // Don't do use_replace since the texture may already no longer have a ref_count.
-                        // However, we *know* that it is currently in use, so the tracker must already know about it.
-                        texture_barriers.extend(
-                            device_tracker
-                                .textures
-                                .change_replace_tracked(
-                                    id::Valid(texture_use.id),
-                                    TextureSelector {
-                                        levels: range.mip_range.clone(),
-                                        layers: range.layer_range.clone(),
-                                    },
-                                    hal::TextureUses::COPY_DST,
-                                )
-                                .map(|pending| pending.into_hal(texture)),
-                        );
-
-                        collect_zero_buffer_copies_for_clear_texture(
-                            &texture.desc,
-                            device.alignments.buffer_copy_pitch.get() as u32,
-                            range.mip_range.clone(),
-                            range.layer_range.clone(),
-                            &mut zero_buffer_copy_regions,
-                        );
-                    }
-
-                    if !zero_buffer_copy_regions.is_empty() {
-                        debug_assert!(texture.hal_usage.contains(hal::TextureUses::COPY_DST),
-                            "Texture needs to have the COPY_DST flag. Otherwise we can't ensure initialized memory!");
-                        unsafe {
-                            // TODO: Could safe on transition_textures calls by bundling barriers from *all* textures.
-                            // (a bbit more tricky because a naive approach would have to borrow same texture several times then)
-                            self.encoder
-                                .transition_textures(texture_barriers.into_iter());
-                            self.encoder.copy_buffer_to_texture(
-                                &device.zero_buffer,
-                                raw_texture,
-                                zero_buffer_copy_regions.into_iter(),
-                            );
-                        }
+                            },
+                        ));
                     }
                 }
             }
+
+            // TODO: Could we attempt some range collapsing here?
+            for range in ranges.drain(..) {
+                clear_texture(
+                    &texture_use.id,
+                    &*texture,
+                    range,
+                    &mut self.encoder,
```

Comment:
given how the encoder is always passed mutably to clear texture, and that it's effectively an operation being encoded, like many other, would it make sense to put it as a method? i.e. `self.encoder.clear_texture(...)`. No strong preference here, just throwing unsolicited ideas around :)